scripts/FigS2B_adt-plots.R

```diff
@@ -0,0 +1,143 @@
+filtered_sce <- readr::read_rds(filtered_sce_file)
+
+processed_sce_file <- file.path(data_dir, "SCPCL000290_processed.rds")
+processed_sce <- readr::read_rds(processed_sce_file)
+
+output_plot_file <- here::here("figures", "pngs", "FigS2B_adt-plots.png")
+
+# Filtered ADT plot ------------------------------------------------------------
+
+# grab coldata from filterd object 
+filtered_coldata_df <- colData(filtered_sce) |>
+  as.data.frame() |>
+  tibble::rownames_to_column("barcode")
+
+filter_levels <- c("Keep", "Filter (RNA & ADT)", "Filter (RNA only)", "Filter (ADT only)")
+filtered_coldata_df <- filtered_coldata_df |>
+  # add column to represent filtering on both RNA and ADT counts
+  dplyr::mutate(filter_summary = dplyr::case_when(
+    scpca_filter == "Keep" & adt_scpca_filter == "Keep" ~ filter_levels[1],
+    scpca_filter == "Remove" & adt_scpca_filter == "Remove" ~ filter_levels[2],
+    scpca_filter == "Remove" & adt_scpca_filter == "Keep" ~ filter_levels[3],
+    scpca_filter == "Keep" & adt_scpca_filter == "Remove" ~ filter_levels[4],
+    TRUE ~ "Error"
+  ),
+  # make sure Keep is first 
+  filter_summary = forcats::fct_relevel(filter_summary, "Keep"),
+  keep_column = ifelse(
+    stringr::str_starts(filter_summary, "Keep"),
+    "Keep",
+    "Remove"
+  ))
+
+# faceted plot to show cells belonging to each filter group
+filtered_plot <- ggplot(filtered_coldata_df, aes(x = detected, y = subsets_mito_percent, color = keep_column)) +
+  geom_point(alpha = 0.5, size = 1) +
+  labs(
+    x = "Number of genes detected",
+    y = "Mitochondrial percentage"
+  ) +
+  facet_wrap(vars(filter_summary)) +
+  theme(
+    plot.title = element_text(hjust = 0.5),
+    legend.position = "none"
+  )
+
+# Top ADTs ---------------------------------------------------------------------
+
+# Calculate variance for each ADT
+adt_var <- altExp(processed_sce) |>
+  logcounts() |>
+  apply(1, var, na.rm = TRUE)
+
+# Get the top 4
+top_adts <- adt_var[order(adt_var, decreasing = TRUE)[1:4]] |>
+  names()
+
+# ADT expression density -------------------------------------------------------
+
+# grab expression for top ADTs from counts
+var_adt_exp_df <- logcounts(altExp(processed_sce))[top_adts, ] |>
+  # as.matrix needs to come _first_
+  as.matrix() |>
+  t() |>
+  as.data.frame() |>
+  tibble::rownames_to_column("barcode") |>
+  # combine all ADTs into a single column for  faceting
+  tidyr::pivot_longer(
+    !barcode,
+    names_to = "ADT",
+    values_to = "adt_expression"
+  )
+
+adt_density_plot <- ggplot(var_adt_exp_df, aes(x = adt_expression, fill = ADT)) +
+  geom_density() +
+  facet_wrap(vars(ADT), nrow = 2) +
+  labs(x = "Log-normalized ADT expression",
+       y = "Density") +
+  theme(legend.position = "none")
+
+# ADT expression UMAP ----------------------------------------------------------
+
+# create data frame of UMAPs and expression
+umap_df <- scuttle::makePerCellDF(processed_sce) |>
+  tibble::rownames_to_column("barcode") |>
```

Comment:
I feel like we might already actually have a column `barcodes` in the colData so you might be able to skip this, but definitely check if I'm right.